server/repository/src/db_diesel/document_registry_row.rs

```diff
@@ -56,14 +57,26 @@ impl<'a> DocumentRegistryRowRepository<'a> {
         DocumentRegistryRowRepository { connection }
     }
 
-    pub fn upsert_one(&self, row: &DocumentRegistryRow) -> Result<(), RepositoryError> {
+    pub fn upsert_one(&self, row: &DocumentRegistryRow) -> Result<i64, RepositoryError> {
         diesel::insert_into(document_registry_dsl::document_registry)
             .values(row)
             .on_conflict(document_registry_dsl::id)
             .do_update()
             .set(row)
             .execute(self.connection.lock().connection())?;
-        Ok(())
+        self.insert_changelog(&row.id, RowActionType::Upsert)
+    }
+
+    fn insert_changelog(&self, uid: &str, action: RowActionType) -> Result<i64, RepositoryError> {
+        let row = ChangeLogInsertRow {
+            table_name: ChangelogTableName::Document,
```

Comment:
This table_name is duplicated, so probably mean something's wrong, maybe just need one of these?